### ISSUE TITLE
[Bug] incorrect displaying of recent macros names 

### ISSUE THREAD

**Issue by FreeCAD-Tools** [FreeCAD/FreeCAD#8734]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

0.21 (Development)

### Full version info

```shell
OS: Windows 10 Version 1909
Word size of FreeCAD: 64-bit
Version: 0.21.0.31917 (Git)
Build type: Release
Branch: master
Hash: 4639283b459f9ff9093e289908cc003db8745b82
Python 3.10.9, Qt 5.15.6, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
```

### Subproject(s) affected?

None

### Issue description

File "Test.FC.Macro.FCMacro" shows as simply "Test"

Everything that comes after the dot in the name is not displayed.

![изображение](https://user-images.githubusercontent.com/90561109/222877242-9cb4e56a-a12b-4514-bb58-28875acbad25.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
@Syres916 want to take a stab at this ?

**Comment by Syres916** [comment]
Looks like a really simple change of line 1168 in src\Gui\Action.cpp from:

`        recentFiles[index]->setText(QString::fromLatin1("%1 %2").arg(index+1).arg(fi.baseName()));`

to:

`        recentFiles[index]->setText(QString::fromLatin1("%1 %2").arg(index+1).arg(fi.completeBaseName()));`

If the OP wants to create the PR then fine with me.

**Comment by luzpaz** [comment]
@FreeCAD-Tools want to make a PR ?

**Comment by FreeCAD-Tools** [comment]
@luzpaz @Syres916

PR should be done by the one who solves the problem. In any case, I won't be able to debug this C++ code for checking. But I don't mind if you add me as a co-author to the PR. Just add text below to end of PR message as that do in [https://github.com/shaise/FreeCAD_FastenersWB/pull/249](https://github.com/shaise/FreeCAD_FastenersWB/pull/249) . This should work.

Co-Authored-By: FreeCAD-Tools [90561109+FreeCAD-Tools@users.noreply.github.com](mailto:90561109+FreeCAD-Tools@users.noreply.github.com)

**Comment by donovaly** [comment]
The fix will be in FreeCAD 0.20.3
